re org of the readme (#85)
PegaSysEng/besu-sample-networks: README.md

- [Nodejs](https://nodejs.org/en/download/) and [Truffle](https://www.trufflesuite.com/truffle) if using the DApp

## Example Network Setups
All our documentation can be found on the [Besu documentation site](https://besu.hyperledger.org/Tutorials/Examples/Private-Network-Example/).

> There are multiple examples in this repo, and each has a POW(proof of work) and POA(proof of authority) setup. Each setup 
> comprises a minimum of 4 ethereum nodes with monitoring tools like:

Reviewer: ```suggestion
There are multiple examples in this repo, and each has a Proof Of Work (POW) and Proof of Authority (POA) setup. Each setup 
comprises a minimum of 4 Ethereum nodes with monitoring tools like:
```

Author: :+1:

- [Alethio Lite Explorer](https://besu.hyperledger.org/en/stable/HowTo/Deploy/Lite-Block-Explorer/) to explore blockchain data at the block, transaction, and account level
- [Metrics monitoring](https://besu.hyperledger.org/en/stable/HowTo/Monitor/Metrics/) via prometheus and grafana to give you insights into how the chain is progressing
- Optional [logs monitoring](https://besu.hyperledger.org/en/latest/HowTo/Monitor/Elastic-Stack/) to give you real time logs of the nodes. This feature is enabled with a `-e` flag when starting the sample network

The examples also include architecture diagrams to visually show components. They generally use the POA (IBFT2 algorithm) setup, and to view the architecture diagrams for the POW (ethash) setup please see the `images` folder (where the POA variants have different suffixes). 

> Each section also includes a few use cases personas (intended as guidelines only).

Reviewer: ```suggestion
Each section also includes use case personas (intended as guidelines only).
```

Author: :+1:

 
**To start services and the network:**

`./run.sh` starts all the docker containers in POW mode

`./run.sh -c ibft2` starts all the docker containers in POA mode using the IBFT2 Consensus algorithm

There is an optional `-e` parameter which provides centralised logging functionality via ELK 

**To stop services :**

`./stop.sh` stops the entire network, and you can resume where it left off with `./resume.sh` 

`./remove.sh ` will first stop and then remove all containers and images

### i. POW (ethash) Network  <a name="pow-network"></a>

This is the closest thing to how 'BitCoin' works, where miners create blocks. In the Ethereum space, the 'mainnet' and 'ropsten' public networks are POW based
 
![Image basic_pow](./images/sampleNetworks-pow.png)

Start the network with: 

`./run.sh ` 

Use cases: 

>  - you are learning about how ethereum works 

Reviewer: ```suggestion
 - you are learning about how Ethereum works 
```

Author: :+1:

 - you are looking to create a mainnet or ropsten node but want to see how it works on a smaller scale
 - you are a DApp Developer looking for a robust, simple network to use as an experimental testing ground for POCs. 
 Generally speaking DApp developers prefer the immediate finality that a POA IBFT2 algorithm offer but POW networks work wells too
 
### ii. POA (ethash) Network  <a name="poa-network"></a>
 
![Image basic_poa](./images/sampleNetworks-poa.png)

Start the network with: 

`./run.sh -c ibft2` 

Use cases: 

>  - you are learning about how ethereum works 

Reviewer: ```suggestion
 - you are learning about how Ethereum works 
```

>  - you are looking to create a private eth network

Reviewer: ```suggestion
 - you are looking to create a private Ethereum network
```

![Image dapp](./images/sampleNetworks-dapp.png)

### iv. [POA (IBFT2) Network with ELK for centralised logs] <a name="poa-network-logs"></a>

This is the same as example ii. [POA (IBFT2) Network](#poa-network) but adds in centralized logging via ELK 

![Image basic_elk](./images/sampleNetworks-poa-elk.png)

Start the network with: 

`./run.sh -c ibft2 -e` 

Use cases: 

>  - you are learning about how ethereum works 
>  - you are looking to create a private eth network

Reviewer: ```suggestion
 - you are learning about how Ethereum works 
 - you are looking to create a private Ethereum network
```

 - you are a DevOps engineer or administrator looking to see how the full blockchain works with logging and metrics
 - you are a DApp developer and looking to build on the previous example with the ability to see transaction logs via ELK 

### v. POA (IBFT2) Network with Privacy via Orion <a name="poa-network-privacy"></a>

![Image basic_orion_elk](./images/sampleNetworks-poa-orion-elk.png)

Start the network with: 
`./run-privacy.sh -c ibft2 -e` starts all the docker containers in POA mode using the IBFT2 Consensus algorithm, and also has 3 Orion nodes for privacy 

`./run-privacy.sh` starts all the docker containers in POW mode, and also has 3 Orion nodes for privacy 

Use cases: 

>  - you are learning about how eth works 

Reviewer: ```suggestion
 - you are learning about how Ethereum works 
```

 - you are a user looking to execute private transactions at least one other party

>  - you are looking to create a private eth network with private transactions between two or more parties. The logs make it easy to see whats going on betwen nodes and transactions

Reviewer: ```suggestion
 - you are looking to create a private Ethereum network with private transactions between two or more parties. The logs make it easy to see whats going on between nodes and transactions
```

This can be verified from the `data` field of the `logs` which is `1`.

### vi. POA (IBFT2) Network with On Chain Permissioning <a name="poa-network-permissioning"></a>

This example showcases on chain permissioning by deploying come [smart contracts](https://github.com/PegaSysEng/permissioning-smart-contracts)

![Image basic_permissioning](./images/sampleNetworks-poa-permissioning.png)

Start the network with: 

`./run-permissioning.sh -e` gets the latest smart contract code, compiles the contracts and updates the genesis file with the contract code. Once done it spins up a full network 

`./run-permissioning-dapp.sh -e` With the network up from the previous step, it will migrate the contracts to the network. Once complete, it restarts the blockchain network with permissions enabled so the rules and permissions deployed in the previous step take effect

Open a new tab in your browser and go to `http://localhost:3001` to use the Permissioning DApp 

Use this scenario:

>  - if you are a DevOps engineer or administrator looking to see how the full blockchain works with on chain permissioning and retrictions

Reviewer: ```suggestion
 - if you are a DevOps engineer or administrator looking to see how the full blockchain works with onchain permissioning and restrictions
```

    - `0xc87509a1c067bbde78beb793e6fa76530b6382a4c0241e5e4a9ec0a0f44dc0d3`
    - `0xae6ae8e5ccbfb04590405997ee2d52d2b330726137b875053c36d94e974d162f`

Open a new tab in your browser and go to `http://localhost:3001` to use the Permissioning DApp, where you can allow/disallow nodes from the network

### vii. POA (IBFT2) Network with Ethsigner to sign transactions <a name="poa-network-ethsigner"></a>

![Image ethsigner](./images/sampleNetworks-poa-signer.png)

Start the network with: 

`./run.sh -c ibft2 -s` gets the latest smart contract code, compiles the contracts and updates the genesis file with the contract code. Once done it spins up a full network 

Use this scenario:

>  - if you need to sign transactions with a private key and forward that along to the eth client ie. Besu

Reviewer: ```suggestion
 - if you need to sign transactions with a private key and forward that to the Ethereum client (for example Besu)
```
